code/application/libraries/validation.php

```diff
@@ -803,18 +803,49 @@ public function validate_user_login($params) {
         try {
             $CI = & get_instance();
             $CI->load->library('form_validation');
-            if ($CI->form_validation->required($params['email']) == False) {
-                $result['status'] = 0;
-                $result['errors'][] = "email required";
-            }
-            if ($CI->form_validation->valid_email($params['email']) == False) {
-                $result['status'] = 0;
-                $result['errors'][] = "invalid email Address";
-            }
-            if ($CI->form_validation->required($params['password']) == False) {
-                $result['status'] = 0;
-                $result['errors'][] = "password Required";
+
+
+            
+
+
+
+
+            if (!empty($params['email'])){
+
+
+
+
+                if ($CI->form_validation->required($params['email']) == False) {
+                    $result['status'] = 0;
+                    $result['errors'][] = "email required";
+                }
+                if ($CI->form_validation->valid_email($params['email']) == False) {
+                    $result['status'] = 0;
+                    $result['errors'][] = "invalid email Address";
+                }
+                if ($CI->form_validation->required($params['password']) == False) {
+                    $result['status'] = 0;
```

Comment:
code redundant for password required (written for both cases should be written outside both if and else if)